hera_cal/red_groups.py

```diff
@@ -565,18 +584,42 @@ def keyed_on_bls(self, bls: Sequence[BlLike], inplace: bool = False) -> Redundan
                 obj._reset_ubl(bl, self[bl])
         if not inplace:
             return obj
-        
+
+    def get_reds_in_bl_set(
+        self,
+        bl: BLLike,
+        bl_set: Sequence[BlLike],
+        include_conj: bool = False,
+        include_conj_only_if_missing: bool = False,
+    ) -> set[BlLike]:
+        """Return a list of redundant groups that contain baselines in the given set."""
+        all_red_bls = self[bl]
+
+        out = {key for key in all_red_bls if key in bl_set}
+
+        if include_conj:
+            all_red_bls = self[reverse_bl(bl)]
+            newout = {reverse_bl(key) for key in all_red_bls if (key in bl_set)}
+            if include_conj_only_if_missing:
+                newout = {key for key in newout if reverse_bl(key) not in out}
+            out = out | newout
+
+        return out
+
+
 @attrs.define(frozen=True)
 class BaselineKeyChooser:
     """A callable that chooses a unique key for a redundant group.
-    
+
     The callable takes a list of baselines and returns a unique key for the group.
     """
     bls: Sequence[BlLike] = attrs.field()
     chooser: Callable[[Sequence[BlLike]], BlLike] = attrs.field()
 
     def __call__(self, red: Sequence[BlLike]) -> BlLike:
         filtered_red = [bl for bl in red if bl in self.bls]
-        return self.chooser(filtered_red) if filtered_red else self.chooser(red)
-    
-    
\ No newline at end of file
+
+        if not filtered_red:
+            raise NonExistentBaselineError("No baselines in the group are in the provided set.")
```

Comment:
This is causing `hera_cal/tests/test_datacontainer.py::test_RedDataContainerKeyManipulation` to fail. Do we need to raise an error here rather than just do the previous default behavior?